### ISSUE TITLE
AE Baseline Training Config

### ISSUE THREAD

**Issue by InkosiZhong** [Hhhhhhao/continuous_tokenizer#10]
Hi, thanks for this impressive work!

Recently, I have been trying to use this framework to reproduce the results of the **simplest AE** in the MAETok paper.
However, I found that the training was difficult to converge. 
Can you give me a config file for reference?

Thanks.

This is what I'm currently using:
``` yaml
results_dir: experiments/tokenizer
cloud_save_path: experiments/tokenizer
compile: False 
global_seed: 42
log_every: 50
val_every: 5000
ckpt_every: 10000
mixed_precision: bf16
no_local_save: True

dataset: imagenet
data_path: /ytech_m2v2_hdd/dataset/imagenet-1k/train.csv
val_data_path: /ytech_m2v2_hdd/zhongtianxiong/VAE/continuous_tokenizer/data/vpg_val_psy_img.csv
data_column: image_path
image_size: 256

vq_model: AE-16
base_image_size: 256
num_latent_tokens: 128
num_codebooks: 4
codebook_embed_dim: 32
codebook_l2_norm: True
codebook_show_usage: True
codebook_size: 8192
commit_loss_beta: 0.0
dec_cls_token: False
dec_drop_path_rate: 0.0
dec_patch_size: 16
dec_pretrained: False
dec_tuning_method: full
dec_type: vit
decoder_model: vit_base_patch14_dinov2_movqv2
dropout_p: 0.0
enc_drop_path_rate: 0.0
enc_patch_size: 16
enc_pretrained: False
enc_token_drop: 0.0
enc_token_drop_max: 0.6
enc_tuning_method: full
enc_type: vit
encoder_model: vit_base_patch14_dinov2.lvd142m

repa: False
repa_align: repeat
repa_loss_weight: 0.1
repa_model: vit_large_patch14_dinov2.lvd142m
repa_patch_size: 16
repa_proj_dim: 1024

entropy_loss_ratio: 0.0
kl_loss_weight: 0.0
vq_loss_ratio: 0.0
reconstruction_weight: 1.0
perceptual_weight: 1.0
perceptual_warmup: 10000
disc_weight: 0.2
disc_adaptive_weight: False
use_diff_aug: True 
disc_cr_loss_weight: 4.0
disc_start: 20000
disc_type: dino
disc_loss: hinge
gen_loss: hinge
lecam_loss_weight: 0.001

vq_mean: 0.0
vq_std: 1.0
z_channels: 256
rope_mixed: True
rope_theta: 10.0
tau: 0.07
to_pixel: linear
use_ape: False
use_rope: True

epochs: 50
lr: 1.0e-4
optim: adamw
lr_scheduler: cosine
weight_decay: 0.0001
beta1: 0.9
beta2: 0.95
max_grad_norm: 1.0
global_batch_size: 32
```

**Comment by Hhhhhhao** [comment]
Thanks for your interest!

I haven't update the training code of MAETok in this repo really, the current training code is mainly for SoftVQ. 
Mask modeling and auxiliary decoders are not included in the training code yet. I will try to update the training code soon, but they should be quite straightforward to implement actually. 

Also when you say 'difficult to converge', can you be more specific and post some loss curve here? So I can help you see what could be the potential bug.

**Comment by InkosiZhong** [comment]
I am not trying to implement MAETok, but just the original AE.
Here are the loss curves (yellow for AE-16 and gray for SoftVQ-L).

<img width="256" alt="Image" src="https://github.com/user-attachments/assets/d22d2143-3938-4424-9ef0-0c443e31d4a3" />
<img width="256" alt="Image" src="https://github.com/user-attachments/assets/d15d0a42-c8c5-47da-a3f6-0488ac5f09ef" />
<img width="256" alt="Image" src="https://github.com/user-attachments/assets/a02ddc4b-d624-4b53-a615-28ec804249e8" />

And this is the visualization.
<img width="256" alt="Image" src="https://github.com/user-attachments/assets/873a8f53-59e9-4dc9-b7a2-89bd58aeb3c3" />

**Comment by Hhhhhhao** [comment]
Looks like it is due to the adversarial loss. You can try to turn on 'disc_adaptive_weight: False' and reduce the perceptual loss weight to 0.5.

Also make the discriminator start larger usually helps

**Comment by InkosiZhong** [comment]
Thanks, I will try it.
